src/util/cmdlineargs.cpp

```diff
@@ -36,92 +38,198 @@ CmdlineArgs::CmdlineArgs()
-        } else if (logLevel == QLatin1String("info")) {
-            *pLogLevel = mixxx::LogLevel::Info;
-        } else if (logLevel == QLatin1String("warning")) {
-            *pLogLevel = mixxx::LogLevel::Warning;
-        } else if (logLevel == QLatin1String("critical")) {
-            *pLogLevel = mixxx::LogLevel::Critical;
-        } else {
-            return false;
-        }
-        return true;
+bool parseLogLevel(
+        const QString& logLevel,
+        mixxx::LogLevel* pLogLevel) {
+    if (logLevel.compare(QLatin1String("trace"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Trace;
+    } else if (logLevel.compare(QLatin1String("debug"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Debug;
+    } else if (logLevel.compare(QLatin1String("info"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Info;
+    } else if (logLevel.compare(QLatin1String("warning"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Warning;
+    } else if (logLevel.compare(QLatin1String("critical"), Qt::CaseInsensitive) == 0) {
+        *pLogLevel = mixxx::LogLevel::Critical;
+    } else {
+        return false;
     }
+    return true;
 }
+} // namespace
+
+bool CmdlineArgs::Parse(const QStringList& arguments) {
+    QCommandLineParser parser;
+    parser.setApplicationDescription(QCoreApplication::translate("main",
+            "Mixxx is an open source DJ software. For more information, see "
+            "https://manual.mixxx.org/2.3/chapters/appendix.html#command-line-options)"));
+    parser.setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
+
+    // add options
+    const QCommandLineOption fullscreen(QStringList() << "f"
+                                                      << "fullScreen",
+            QCoreApplication::translate("main", "Starts Mixxx in full-screen mode"));
+    parser.addOption(fullscreen);
+
+    const QCommandLineOption locale(QStringLiteral("locale"),
+            QCoreApplication::translate("main",
+                    "Use a custom locale for loading translations. (e.g "
+                    "'fr')"));
+    parser.addOption(locale);
+
+    // An option with a value
+    const QCommandLineOption settingsPath(QStringLiteral("settingsPath"),
+            QCoreApplication::translate("main",
+                    "Top-level directory where Mixxx should look for settings. "
+                    "Default is:") +
+                    getSettingsPath().toLocal8Bit().constData(),
+            QStringLiteral("settingsPath"));
+    parser.addOption(settingsPath);
+
+    const QCommandLineOption pluginPath(QStringLiteral("pluginPath"),
+            QCoreApplication::translate("main",
+                    "Top-level directory where Mixxx should look for sound "
+                    "source plugins in addition to default locations."),
+            QStringLiteral("pluginPath"));
+    parser.addOption(pluginPath);
```

Comment:
This is obsolete since 2.3. Please remove it here. We should remove it from the 2.3 branch too.

Reply:
#3651

Reply:
There are conflicts now that #3651 was merged.